### ISSUE TITLE
Moving to Python 3.X

### ISSUE THREAD

**Issue by xgarrido** [davidfokkema/artist#13]
Is there any plan to move to Python 3.X ? My current fork has done the minimal changes to be Python 3.X compatible. That's not a big deal and may be worth it.

**Comment by 153957** [comment]
We have no plan to move entirely to Python 3.X and drop Python 2 support.
But perhaps we can make it both 2.7.X and 3.X compatible, I would not be opposed to that.

**Comment by davidfokkema** [comment]
In the HiSPARC group, we're currently using Python 2.7 exclusively (simulation, analysis, data storage backend, etc. etc.). We won't be upgrading anytime soon, unfortunately.

That being said, if we can support both Python 2.x and Python 3.x I'm happy to merge a pull request. Are there any pitfalls?

**Comment by xgarrido** [comment]
As far as I remember, there was only the use of ```zip_longest``` module (not anymore ```izip_longest```) that introduces such change

```
 series_x, series_y = list(zip(*data))[:2]
```

in place of  
```
series_x, series_y = zip(*data)[:2]
```

**Comment by 153957** [comment]
And some print statements need to be replaced by the print function, which is supported by both 3.x and 2.7.

**Comment by davidfokkema** [comment]
Well, if that's all, I'm all for it. @xgarrido are you interested in creating a pull request? I don't know if you've done other work on your fork. I'd be happy to consider other changes as well, but if possible in another pull request.

**Comment by xgarrido** [comment]
@davidfokkema I can do it but I'm quite busy now and I'm not a Python coder (actually I learn a lot from your code). Just to make things clear, the idea will be to support both Python 3.X and Python 2.7 ? I will try to create a clean branch with only Python 3.X fixes.

@153957 I only find a ```print``` function in the demo. As you mentionned, the ```None/int``` comparison also had to be fixed.

**Comment by 153957** [comment]
Indeed, we want to support both Python 2.7 and 3.x.
So if you make fixes for Python 3.x try to make sure they do not break Python 2.7 compatibility.

For the `izip_longest` you can change the import to:
``` python
try:
    # Python 2
    from itertools import izip_longest
except ImportError:
    # Python 3
    from itertools import zip_longest as izip_longest
```

**Comment by davidfokkema** [comment]
@xgarrido I see you have a number of commits, some fixing python 3 support, some adding features. If you're not in a hurry, we can look over them and maybe merge your whole branch. Or maybe we'll cherry pick your python 3 commits and create a branch from that.

**Comment by xgarrido** [comment]
I have started the process to make artist both Python 2.7/3.X compatible. I am testing each changes (not so much) against Python 2.7 and Python 3.4 and I started to pull some changes (see #20 but as @153957 underlined there is still some places to change).

The only piece of code that does not satisfy me is this one xgarrido@098bc5c2ff3eadd527d4d8b48427c157901a51bb since Python 3.X does not implicitly converts None to float type. Is it ok for you or do you have any better idea to deal with this issue ?

**Comment by davidfokkema** [comment]
I see what you mean. I'll comment on that commit. BTW: thanks for taking the time to work on this!

**Comment by 153957** [comment]
Fixed in 55db87b09c511a6eb5c
